feat(mesh): profile page

LinHeMa opened the pull request:
# Profile Page Implementation

## Features
- Implemented profile page UI
- Introduced dual-route access:
  - `profile/member`
  - `profile/publisher`
- User types:
  - Member route: handles both `visitor` and `member` (self) views
  - Current workaround: Utilizing query string `?user={itemsID}` for user identification


## Work in Progress (WIP)
- [x] Refactor comment line clamp logic:
  - Extract logic from UI component
  - Implement as a custom hook for improved reusability
- [x] Optimize empty state UI and logic

## Technical Considerations
- The separation of user information components is designed to manage increasing complexity of user authorities
- Current routing solution is temporary, pending further development(login and register)

## Next Steps
- turn mobile header into correct one

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207550327352689

LinHeMa committed:
feat(profie): gql and fetch

LinHeMa committed:
feat(profie): gql and fetch

LinHeMa committed:
tmp

LinHeMa committed:
tmp

LinHeMa committed:
feat(mesh/profile): mobile layout without comment

LinHeMa committed:
feat(mesh/profile): mobile layout without comment

LinHeMa committed:
tmp(mesh/profile): comment block

LinHeMa committed:
tmp(mesh/profile): comment block

LinHeMa committed:
feat(mesh/profile): mobile done

LinHeMa committed:
feat(mesh/profile): mobile done

LinHeMa committed:
feat(mesh/profile): tablet, pc layout

LinHeMa committed:
feat(mesh/profile): tablet, pc layout

LinHeMa committed:
feat(mesh/profile): UI done

LinHeMa committed:
feat(mesh/profile): UI done

LinHeMa committed:
fix(mesh/profile): delete TODO

LinHeMa committed:
fix(mesh/profile): delete TODO

LinHeMa committed:
doc(mesh/profile): gql code gen

LinHeMa committed:
doc(mesh/profile): gql code gen

LinHeMa committed:
refactor(mesh/profile): separate logic from UI

LinHeMa committed:
refactor(mesh/profile): separate logic from UI

LinHeMa committed:
refactor(mesh/profile): common logic separate

LinHeMa committed:
refactor(mesh/profile): common logic separate

LinHeMa committed:
refactor(mesh/profile): common UI separate

LinHeMa committed:
refactor(mesh/profile): common UI separate

LinHeMa committed:
feat(mesh/profile): empty status

LinHeMa committed:
feat(mesh/profile): empty status

LinHeMa committed:
fix(mesh/profile): storyid

LinHeMa committed:
fix(mesh/profile): storyid

LinHeMa committed:
fix(mesh/profile): delete log

LinHeMa committed:
fix(mesh/profile): delete log

tyc67 reviewed (changes requested):
- `type.ts` , `utils.ts`  參考 [folder structure convention](https://paper.dropbox.com/doc/Convention--CSIFAAQU76EqjKKvb3UjEws1Ag-yDGKaFsVLerISYtxFlM9e) 
- text typography 參考 `styles/global.css` 
- button 參考 `components/button.tsx` 
- 建議 Component 架構, 增加可讀性
```jsx
<ProfileHeader/>
<Profile>
  <UserIntro/>
  <ButtonLists/>
  <UserNumbers/>
  <TabLists/>
  <StoryList/>
<Profile/>
```

caesarWHLee commented on packages/mesh/app/profile/[type]/[id]/page.tsx:
publisher 和 member 的 profile 建議拆成兩個 folder (route) 就好，避免不必要的複雜度，像是目前要看 member 的 profile 的話其實 url 帶 http://localhost:3000/profile/eee/19?user=19 或是 http://localhost:3000/profile/aaaaa/19?user=19 其實都可以看到某個 member 的 profile，邏輯上不太正確。

建議可以跟 PM 確認一下兩個頁面的 route，看是要 profile/member/[id], profile/publisher/[id] 還是 profile/[memberId], profile/publisher/[id] 切成兩個 page.tsx 然後各自的 components 放到對應的 folder 中會比較好理解維護

caesarWHLee commented on packages/mesh/app/profile/_components/member-profile.tsx:
```diff
@@ -0,0 +1,58 @@
+import {
+  GetMemberProfileDocument,
+  GetMemberProfileQuery,
```
加上 type keyword 來區別 types 和 components

caesarWHLee commented on packages/mesh/graphql/query/profile.gql:
gql files 目前的 conventions 是依照 gql schema 來放，以 `GetMemberProfile`  這個 query 來說要拿的是 `member` 這個 list，所以應該移到 member.gql 裡面管理

命名公式：[listName].gql

caesarWHLee commented on packages/mesh/graphql/query/publisherProfile.gql:
gql files 目前的 conventions 是依照 gql schema 來放，以 `GetPublisherProfile`  這個 query 來說要拿的是 `publisher` 這個 list，所以應該移到 publisher.gql 裡面管理

命名公式：[listName].gql

caesarWHLee commented on packages/mesh/app/profile/_components/member-profile.tsx:
```diff
@@ -0,0 +1,58 @@
+    memberId: visitID,
+  })
+  if (!response) {
+    return <div>Error</div>
```
建議加上 TODO: comment 以利日後處理

caesarWHLee commented on packages/mesh/app/profile/_components/member-profile.tsx:
```diff
@@ -0,0 +1,58 @@
+  }
+  const userData = response.member
+  if (!userData) {
+    return <div>Error</div>
```
建議加上 TODO: comment 以利日後處理

caesarWHLee commented on packages/mesh/app/profile/_components/member-profile.tsx:
```diff
@@ -0,0 +1,58 @@
+    : 'min-h-screen-without-header-nav sm:min-h-sm-screen-without-header-footer'
+
+  return (
+    <div className={`flex ${contentHeight}  flex-col bg-white`}>
```
Page 的 root element 要使用 `main`

caesarWHLee commented on packages/mesh/app/profile/_components/member-profile.tsx:
```diff
@@ -0,0 +1,58 @@
+  const hasContent = picksData.length > 0 || bookmarkData.length > 0
+  const contentHeight = hasContent
+    ? 'h-full'
+    : 'min-h-screen-without-header-nav sm:min-h-sm-screen-without-header-footer'
```
之前的 layout 沒寫好所以會需要 chilrdren 各自計算高度來填滿螢幕高，目前修改一版在 [PR](https://github.com/readr-media/Sachiel/pull/791)，這個版本下就不需要再另外計算高度，可以在 root elemnt 上面設置 className="grow" 就會讓 root element 的高長滿畫面

同樣的，再下一層的 component 像是 profile 沒有精選的狀況下要顯示 `這個人還沒有精選新聞` 的區塊，也可以一路設定 parent `flex`, child `grow` 的方式來填滿螢幕

caesarWHLee reviewed (commented):
先就目前看到的進度給回饋

LinHeMa committed:
refactor(mesh/profile): router split

LinHeMa committed:
refactor(mesh/profile): router split

LinHeMa committed:
refactor(mesh/profile): change GQL type

LinHeMa committed:
refactor(mesh/profile): change GQL type

LinHeMa committed:
refactor(mesh/app): layout

LinHeMa committed:
refactor(mesh/app): layout

LinHeMa committed:
refactor(mesh/profile): layout

LinHeMa committed:
refactor(mesh/profile): layout

LinHeMa committed:
feat(mesh/profile): layout

LinHeMa committed:
feat(mesh/profile): separate layout

LinHeMa committed:
refactor(mesh/profile): change to common type

LinHeMa committed:
feat(mesh/profile): global css add classes

LinHeMa committed:
Merge branch 'feat-profile-PR' of https://github.com/LinHeMa/Sachiel into feat-profile-PR

LinHeMa committed:
Merge branch 'dev' of https://github.com/readr-media/Sachiel into feat-profile-PR

LinHeMa committed:
refactor(mesh/profile): change btn to common

LinHeMa committed:
refactor(mesh/profile): delete unused

1. change member id to custom id
2. delete comment
3. delete layout 'use client'

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/page.tsx:
```diff
@@ -0,0 +1,60 @@
+        name={userName}
+        avatar={userLogo}
+        intro={userIntro}
+        followerCount={userFollowerCount || 0}
```
和 Ln.41 一樣?

tyc67 commented on packages/mesh/app/layout.tsx:
```diff
@@ -19,7 +19,7 @@ export default function RootLayout({
   return (
     <html lang="zh-Hant" className={notoSans.className}>
-      <body className="min-h-screen">{children}</body>
+      <body className="flex min-h-screen grow flex-col">{children}</body>
```
應該是加在Page layer Layout ?

tyc67 replied:
RootLayout 應該不會給 flex, 由各 Page layout 控制

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,75 @@
+    return userStatusList
+  }
+
+  const emptyStatusMaxHeight = 'max-h-[calc(100%_-_152px)]'
```
沒有condition operator, 是否直接寫在 classname?

tyc67 commented on packages/mesh/app/profile/member/[userId]/layout.tsx:
```diff
@@ -0,0 +1,23 @@
+        <ProfileFooter />
+      </Body>
+      <Nav />
+    </div>
```
- `ProfileLayout` 應該只需要一個在 `profile/layout`
- `<ProfileHeader />`,`<Body>`,`<ProfileFooter />`,應該可以直接寫在layout裡面
- `<main>` 目前用於`<Page/>`

tyc67 commented on packages/mesh/app/profile/_components/body.tsx:
```diff
@@ -0,0 +1,7 @@
+export default function Body({ children }: { children: React.ReactNode }) {
+  return (
+    <main className="flex min-h-screen flex-col bg-white pb-[theme(height.nav.default)] pt-[theme(height.header.default)] sm:pb-0 sm:pl-[theme(width.nav.sm)] sm:pt-[theme(height.header.sm)] md:pl-[theme(width.nav.md)] xl:pl-[calc((100vw-theme(width.maxContent))/2+theme(width.nav.xl))]">
```
可以用 `global.css` 的 `primary-container`

tyc67 commented on packages/mesh/styles/global.css:
```diff
@@ -75,14 +79,18 @@
+  .profile-title-3 {
+    @apply text-xl font-medium leading-7;
+  }
+
```
[figma guideline](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=3889-258100&t=72HNFgKG87Mm4ISv-4) 似乎沒有這個, 可能需要和設計確認

tyc67 commented on packages/mesh/styles/global.css:
```diff
@@ -59,6 +59,10 @@
+  .caption-3 {
+    @apply text-xs font-normal;
+  }
+
```
[figma guideline](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=3889-258100&t=72HNFgKG87Mm4ISv-4) 似乎沒有這個, 可能需要和設計確認

tyc67 commented on packages/mesh/components/button.tsx:
```diff
@@ -42,6 +42,7 @@ export default function Button({
     sm: 'button h-8 px-3 py-1 rounded', // all Button Mini
     md: 'button-large h-[38px] px-5 py-2 rounded', // Secondary Button
     'md-100': 'button-large h-[38px] px-4 py-2 rounded-[100px]', // Lightbox Chip
+    'md-fixed-w': 'button-large h-[38px] px-4 py-2 rounded-md sm:w-[180px]', // Lightbox Chip
```
應該是用md, 由外層控制寬度?

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/_component/profile-button-list.tsx:
```diff
@@ -0,0 +1,29 @@
+  buttonList: ProfileButton[]
+}
+
+const ProfileButtonLIst: React.FC<ProfileButtonLIstProps> = ({
```
- ProfileButtonL"i"st
- 看起來只有color 不同, 應該不需要兩個 `profile-button-list.tsx`

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,75 @@
+  followerCount,
+  storyData,
+}) => {
+  const [_, setCategory] = useState<TabCategory>(TabCategory.publish)
```
publisher-page 的 Tab 只有"報導", 應該不需要這個 state ?

tyc67 commented on packages/mesh/utils/profile-tab.ts:
```diff
@@ -0,0 +1,32 @@
+export const tabFilter = (userType: string) => {
+  switch (userType) {
+    case 'publisher':
+      return createTabFilter(PUBLISHER_TAB_KEYS)
```
Ln26, Ln28, 是否直接 inline 寫即可? 只有 [FOLLOWER,SPONSORED] 和 [PICK,FOLLOWER,FOLLOWING] 兩種情境的話, 是不是不需用到 `Set` 和 `createTabFilter`

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/_component/user-profile.tsx:
```diff
@@ -0,0 +1,47 @@
+        >
+          <Image
+            alt={`${name}'s avatar`}
+            src={avatar || '/images/default-avatar-image.png'}
```
是否可以用 `icon-avatar-default` ?

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,75 @@
+      >
+        <UserProfile
+          name={name}
+          pickCount={pickCount || 0}
```
如果移至 Page 做 Check, Page 應該就不用 `pickCount={null}` ?

tyc67 commented on packages/mesh/app/profile/publisher/[userId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,75 @@
+      { key: TabKey.PICK, value: pickCount },
+      { key: TabKey.FOLLOWER, value: followerCount },
+      // TODO: sponsored value need api
+      { key: TabKey.SPONSORED, value: '9999次' },
```
可以考慮 naming
```
  const userStatusList = [
    { tab: TabKey.PICK, count: pickCount },
    { tab: TabKey.FOLLOWER, count: followerCount },
    { tab: TabKey.SPONSORED, count: '9999次' ,
  ];
```

tyc67 reviewed (commented):
- 依照 design system, 應該不會有 `leading` 的 classname
- publisher 和 member 的 `_component` 看起來有多個一樣的component, 應該可以共用, 未來不需要維護兩套 css ? 
- `alt="alt"` 也許可以給一個可以辨識的message?

LinHeMa committed:
refactor(profile): merge layout

caesarWHLee commented on packages/mesh/graphql/query/publisher.gql:
```diff
@@ -8,3 +8,54 @@ query Publishers {
   }
 }
+
+query GetPublisherProfile($memberId: ID) {
```
這裡應該叫 publisherId?

caesarWHLee commented on packages/mesh/graphql/query/publisher.gql:
```diff
@@ -8,3 +8,54 @@ query Publishers {
+    followerCount
+    description
+  }
+  stories(where: { source: { id: $memberId } }) {
```
1. syntax error，stories 的 where condition 應該要用 `{equals: $memberId }` 包起來
```
stories(where: { source: { id: {equals: $memberId} } }) {
```

建議之後 gql 的 query 可以寫完丟到 playground 或直接在 playground 撰寫，可以直接查看文件：
![image](https://github.com/readr-media/Sachiel/assets/104348940/a4668a91-3798-41e1-b4d8-2f9fa379c363)

2. stories 應該要加上 take，可以和 PM 討論每頁的數量，或是至少開發上先給一個定值，目前 publisherId = 1 的狀況下會有一千多筆資料，造成 gql server response time 過久
 
3. 承接上一點，新增 storiesCount (where: { source: { id: {equals: $memberId} } }) (平行 stories) 來拿取總 stories 數量，之後用於 infinite-scroll 上

LinHeMa replied:
目前跟PM討論take 20

caesarWHLee commented on packages/mesh/graphql/query/publisher.gql:
```diff
@@ -8,3 +8,54 @@ query Publishers {
+    logo
+    followerCount
+    description
+  }
```
新增 `storiesCount (where: { source: { id: {equals: $memberId} } })` (平行下面 stories) 來拿取總 story 數量，之後用於 infinite-scroll 上

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,165 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+    followingCount
+    followerCount
+
+    picks: pick(where: { kind: { equals: "read" } }) {
```
1. is_active 需要是 true `pick(where: { kind: { equals: "read" }, is_active: {equals: true} })`
2. 加上 take (同 publiser.gql comment)
3. 加上 picksCount (同 publiser.gql comment)

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,165 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+      kind
+    }
+
+    books: pick(where: { kind: { equals: "bookmark" } }) {
```
1. is_active 需要是 true `pick(where: { kind: { equals: "bookmark" }, is_active: {equals: true} })`
2. 加上 take (同 publiser.gql comment)
3. 加上 booksCount (同 publiser.gql comment)

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,165 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+    followingCount
+    followerCount
+
+    picks: pick(where: { kind: { equals: "read" } }) {
```
1. is_active 需要是 true `pick(where: { kind: { equals: "read" }, is_active: {equals: true} })`
2. 加上 take (同 publiser.gql comment)
3. 加上 picksCount (同 publiser.gql comment)

LinHeMa committed:
refactor(profile): delete unused

caesarWHLee commented on packages/mesh/app/profile/publisher/[userId]/page.tsx:
`publisher/[userId]` > `publisher/[publisherId]`

caesarWHLee commented on packages/mesh/app/profile/publisher/[userId]/page.tsx:
```diff
@@ -0,0 +1,60 @@
+  }
+}
+const page = async ({ params }: PageProps) => {
+  const userId = params.userId
```
`userId` > `publisherId`

caesarWHLee commented on packages/mesh/app/profile/member/[userId]/page.tsx:
`userId` > `customId`

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,165 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
   }
 }
+
+query GetMemberProfile($memberId: String) {
```
`$memberId` >`customId`

LinHeMa committed:
refactor(profile): delete null props

LinHeMa committed:
refactor(profile): a11y

caesarWHLee commented on packages/mesh/app/profile/member/[userId]/_components/user-profile.tsx:
```diff
@@ -0,0 +1,47 @@
+          />
+        </div>
+        <div className="flex flex-col justify-center gap-1">
+          <p className="profile-title sm:profile-title-3 text-primary-700">
```
這邊應該是 `sm:title-1` 不需要另外新增 class `.profile-title-3`

目前網站所有字體設定都以對照[設計稿](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=3889-258100&t=rG9OkOG5IC6pStky-4)設定對應 className

元件的字體可以在 figma 上的這邊看到對應的 className，格式是 lower case kebab case
![截圖 2024-07-09 中午12 01 45](https://github.com/readr-media/Sachiel/assets/104348940/1903b1d7-1ea1-4890-b5da-506909a87bd3)

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,91 @@
+              className="object-cover"
+            />
+          </div>
+          <p className="caption-3 leading-[18px] text-primary-500">
```
這裡應該是 `caption-1`，見 `user-profile.tsx` 的 comment

![image](https://github.com/readr-media/Sachiel/assets/104348940/c204f1cc-68c3-48e4-b939-3965e7c5ba70)

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,91 @@
+            {displayTimeFromNow(data.createdAt)}
+          </p>
+          <Icon iconName="icon-dot" size="s" />
+          <button className="caption-3 leading-[18px] text-primary-500">
```
這裡應該是 `caption-1`，見 `user-profile.tsx` 的 comment

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,91 @@
+          </button>
+        </div>
+        <div className="flex items-center justify-end">
+          <p className="caption-3 leading-[18px] text-primary-600">
```
這裡應該是 `caption-1`，見 `user-profile.tsx` 的 comment

LinHeMa committed:
refactor(profile): css

LinHeMa committed:
refactor(profile): change route var

LinHeMa committed:
refactor(profile): remove unneccessory class

LinHeMa committed:
refactore(profile): change gql fetch method and style

LinHeMa committed:
refactor(profile): delete console

LinHeMa committed:
refactor(profile): delete all leading and use shared className

LinHeMa committed:
refactor(profile): change to design system

LinHeMa commented:
> * 依照 design system, 應該不會有 `leading` 的 classname
> * publisher 和 member 的 `_component` 看起來有多個一樣的component, 應該可以共用, 未來不需要維護兩套 css ?
> * `alt="alt"` 也許可以給一個可以辨識的message?


第一點與第三點已解決，第二點的話當初是為了考慮到不同的身份各自擁有自己的component，未來如果有需求的話可以考慮整併，目前先維持寫法。🙏

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,185 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+      urlOriginal
+    }
+    intro
+    pickCount(where: { kind: { equals: "read" }, is_active: { equals: true } })
```
align alias picks to `picksCount`

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,185 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+      urlOriginal
+    }
+    intro
+    pickCount(where: { kind: { equals: "read" }, is_active: { equals: true } })
```
align alias picks to `picksCount`

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/user-profile.tsx:
```diff
@@ -0,0 +1,45 @@
+          />
+        </div>
+        <div className="flex flex-col justify-center gap-1">
+          <p className="profile-title  text-primary-700">{name}</p>
```
`sm:title-1` 沒補到

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/page.tsx:
```diff
@@ -0,0 +1,97 @@
+    'books' in userData && userData.books ? userData.books : []
+
+  return (
+    <>
```
用 `<main className="flex grow flex-col">` 取代讓每個頁面都有 main tag

caesarWHLee commented on packages/mesh/app/profile/publisher/[publisherId]/page.tsx:
```diff
@@ -0,0 +1,59 @@
+  const storyData = response.stories || []
+
+  return (
+    <main className="flex h-full grow flex-col">
```
應該不用 `h-full` ?

caesarWHLee commented on packages/mesh/app/profile/_components/comment/index.tsx:
目前確定不用 `compoundComponet` 的情況下應該不用多這一層的 container，可以把對應的 type 搬到 `comment.tsx` 之中，另外 useCommentLogic.tsx 建議直接搬到 package root 的 hooks 中，naming convention 一樣是 `lower-kebab-case.tsx`

這樣一來的話 comment folder 就不用保留，剩下 `comment.tsx` 就好

caesarWHLee commented on packages/mesh/app/profile/_components/comment/index.tsx:
```diff
@@ -0,0 +1,33 @@
+  avatar: string
+  name: string
+}
+export type CommentData = CommentProps['data']
```
這個 type 好像沒用到

caesarWHLee commented on packages/mesh/app/profile/_components/comment/index.tsx:
```diff
@@ -0,0 +1,33 @@
+'use client'
+
+import { StoryItem } from '../../member/[customId]/_components/article-card'
```
`import { type StoryItem}`

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,89 @@
+  bookmarkData,
+  visitID,
+}) => {
+  const [showData, setShowData] = useState(picksData)
```
`showData` 命名有點太籠統，可以直接叫做 `picksOrBookmarks` 或之類的命名，useState 也應該要把兩個 type 都包含進來，可以考慮用 type alias 定義 `Picks` 和 `Bookmarks`

`useState<Picks | Bookmarks>(picksData)`

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,89 @@
+  followingCount,
+  followerCount,
+  picksData,
+  bookmarkData,
```
bookmarks，複數 ＋ 不用 Data 後綴

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,89 @@
+  pickCount,
+  followingCount,
+  followerCount,
+  picksData,
```
不用 Data 後綴

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/article-card-list.tsx:
```diff
@@ -0,0 +1,70 @@
+}
+const ArticleCardList = ({
+  showData,
+  id,
```
id 命名太籠統，只看 component 不知道是誰的 id

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/article-card-list.tsx:
```diff
@@ -0,0 +1,70 @@
+  name?: string
+}
+const ArticleCardList = ({
+  showData,
```
比照 `member-page.tsx useState` 修改命名

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,87 @@
+}
+
+const Comment: React.FC<CommentProps> = ({
+  data,
```
直接命名叫 comment 就好

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,87 @@
+    <section className="mt-4 flex w-full flex-col gap-2 rounded-md border border-primary-200 bg-primary-100 p-3">
+      <div className="flex items-center justify-between md:hidden">
+        <div className="flex items-center">
+          <div className="mr-2 h-7 w-7 overflow-hidden rounded-full">
```
直接使用共用元件 `<Avatar src={data.member?.avatar} size="m">` ，無需自行帶入 default-avatar-image.png

caesarWHLee commented on packages/mesh/app/profile/publisher/[publisherId]/page.tsx:
```diff
@@ -0,0 +1,59 @@
+    )
+  }
+  const userName = userData.title || '使用者名稱'
+  const userLogo = userData.logo || '/images/default-avatar-image.png'
```
fallback 可以交給 <Avatar/> 元件處理就好，不用重複 hardcode 在這邊

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/user-profile.tsx:
```diff
@@ -0,0 +1,45 @@
+    <>
+      <section className="flex w-full gap-4">
+        <div className="relative aspect-square w-16 overflow-hidden rounded-full sm:w-20">
+          <Image
```
建議改用 `<Avatar>` 共用元件處理，可以擴充元件的 size，新增 xxl 對應 w-20 h-20，RWD 部分這邊可以寫成

```
<Avatar size="xl" src={avatar} extra="sm:hidden"/>
<Avatar size="xxl" src={avatar} extra="hidden sm:inline-block"/>
```

caesarWHLee commented on packages/mesh/app/profile/publisher/[publisherId]/_component/user-profile.tsx:
```diff
@@ -0,0 +1,34 @@
+          className="relative aspect-square w-16 overflow-hidden rounded-lg
+          sm:w-20"
+        >
+          <Image
```
同 `member/../user-profile.tsx` 的 comment 改成使用 `<Avatar>`，不需要重複的 `default-avatar-image.png`

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,87 @@
+  if (width < 960 && !data.content) return <></>
+  return (
+    <section className="mt-4 flex w-full flex-col gap-2 rounded-md border border-primary-200 bg-primary-100 p-3">
+      <div className="flex items-center justify-between md:hidden">
```
建議上方加入 jsx comment `{/* mobile comment info */}` 增加可讀性

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,87 @@
+        } after:body-3 after:absolute after:bottom-0 after:right-1 after:bg-gradient-to-r after:from-transparent after:from-0% after:to-primary-100 after:to-25% after:pl-6 after:text-primary-400 after:content-['...繼續閱讀'] md:after:bottom-[6px]`}
+        onClick={handleToggleClamp}
+      >
+        <div className="mr-2 hidden h-7 min-h-7 w-7 min-w-7 overflow-hidden rounded-full md:flex">
```
建議加上註解 `{/* non-mobile comment avatar */}`

caesarWHLee commented on packages/mesh/app/profile/_components/comment/comment.tsx:
```diff
@@ -0,0 +1,87 @@
+        onClick={handleToggleClamp}
+      >
+        <div className="mr-2 hidden h-7 min-h-7 w-7 min-w-7 overflow-hidden rounded-full md:flex">
+          <Image
```
同 Ln 38 改成使用 `<Avatar>`

caesarWHLee commented on packages/mesh/app/profile/publisher/[publisherId]/_component/story-card-list.tsx:
```diff
@@ -0,0 +1,50 @@
+  return (
+    <>
+      <ul className="max-w-[1120px] bg-primary-700-dark md:grid md:grid-cols-2 md:items-center md:gap-5 md:p-10 lg:grid-cols-3">
+        {storyData.map((story, idx) => {
```
add TODO: comment to remind adding infinite-scroll in the future

caesarWHLee commented on packages/mesh/app/profile/member/[customId]/_components/article-card-list.tsx:
```diff
@@ -0,0 +1,70 @@
+  return (
+    <>
+      <ul className="max-w-[1120px] bg-primary-700-dark md:grid md:grid-cols-2 md:items-center md:gap-5 md:p-10 lg:grid-cols-3">
+        {showData.map((pick, idx) => {
```
add TODO: comment to remind adding infinite-scroll in the future

tyc67 commented on packages/mesh/components/button.tsx:
```diff
@@ -40,7 +40,7 @@ export default function Button({
     xs: 'profile-subtitle h-[36px] px-4 py-2 rounded-[100px]', // Nav Chip, Nav Button-Add
     sm: 'button h-8 px-3 py-1 rounded', // all Button Mini
-    md: 'button-large h-[38px] px-5 py-2 rounded', // Secondary Button
+    md: 'w-full button-large h-[38px] px-5 py-2 rounded', // Secondary Button
```
應該是由外層控制寬度?

tyc67 commented:
> > * 依照 design system, 應該不會有 `leading` 的 classname
> > * publisher 和 member 的 `_component` 看起來有多個一樣的component, 應該可以共用, 未來不需要維護兩套 css ?
> > * `alt="alt"` 也許可以給一個可以辨識的message?
> 
> 第一點與第三點已解決，第二點的話當初是為了考慮到不同的身份各自擁有自己的component，未來如果有需求的話可以考慮整併，目前先維持寫法。🙏

我會建議趁現在記憶尤新的時候調整 Component 設計, 不知道 @caesarWHLee  的看法?

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,51 @@
+    <>
+      <div className="flex max-h-[calc(100%_-_152px)] flex-col items-center px-5 pb-8 pt-6 sm:max-h-full">
+        <UserProfile name={name} avatar={avatar} intro={intro} />
+        <ProfileButtonLIst buttonList={buttonList} />
```
<ProfileButtonL "i" st />

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,51 @@
+  intro,
+  storyData,
+}) => {
+  const userStatusList = [{ key: TabKey.SPONSORED, count: '9999次' }]
```
建議 naming 不要用 `key`, 可以用`tab`,`tabKey`,`tabTitle`.... or 其他, 在 `<UserStatusList />` 的`{key}` 可讀性會比較好

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/tab.tsx:
```diff
@@ -0,0 +1,65 @@
+  }
+  return (
+    <ul
+      className="flex h-[48px] w-full items-center justify-around  border-b border-t border-[rgba(0,9,40,0.1)]
```
border color 應為 design system 的顏色

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/article-card.tsx:
```diff
@@ -0,0 +1,80 @@
+        />
+      </section>
+      <div
+        className={`flex flex-col p-5 after:absolute after:bottom-1 after:h-[1px] after:w-[calc(100%-40px)] after:bg-[rgba(0,0,0,0.1)] md:line-clamp-3 md:pt-[12px] md:after:hidden ${
```
bg-color 應為 design system 的顏色

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/tab.tsx:
```diff
@@ -0,0 +1,32 @@
+  ]
+  return (
+    <ul
+      className="flex h-[48px] w-full items-center justify-around  border-b border-t border-[rgba(0,9,40,0.1)]
```
border color 應為 design system 的顏色

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/article-card-list.tsx:
```diff
@@ -0,0 +1,70 @@
+          return (
+            <li
+              key={pick?.story?.id}
+              className="relative w-full bg-white md:h-full md:rounded-md md:shadow-[0_2px_2px_0px_rgba(0,9,40,0.1)]"
```
shadow color 應為 design system 的顏色

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/story-card-list.tsx:
```diff
@@ -0,0 +1,50 @@
+          return (
+            <li
+              key={story?.id}
+              className="relative h-full w-full bg-white md:rounded-md md:shadow-[0_2px_2px_0px_rgba(0,9,40,0.1)]"
```
shadow color 應為 design system 的顏色

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/story-card-list.tsx:
```diff
@@ -0,0 +1,50 @@
+  const getMessage = (userType: string, category: TabCategory): string => {
+    const key = `${userType}_${category}`
+    return messages[key] || ''
+  }
```
在 `/publisher/[publisherId]` route 底下 userType 應該都是 publisher, 這裡是否還需要做 conditional ?

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,89 @@
+    { key: TabKey.PICK, value: pickCount },
+    { key: TabKey.FOLLOWER, value: followerCount },
+    { key: TabKey.FOLLOWING, value: followingCount },
+  ]
```
建議 naming 不要用 key 和 value

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,89 @@
+      <div className="flex max-h-[calc(100%_-_152px)] flex-col items-center px-5 pb-8 pt-6 sm:max-h-full">
+        <UserProfile
+          name={name}
+          pickCount={pickCount || 0}
```
`<Page/ >` 已經做null check, 這裏應該不用在做一次

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/user-status-list.tsx:
```diff
@@ -0,0 +1,24 @@
+
+const UserStatusList = ({ userStatusList }: UserStatusListProps) => (
+  <ul className="mt-6 flex w-full items-center justify-center gap-6 sm:mt-4 sm:justify-start">
+    {userStatusList.map(({ key, value }) => (
```
建議 naming 不要用 key , value。 不利於可讀性及維護性

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/article-card.tsx:
```diff
@@ -0,0 +1,126 @@
+        />
+      </section>
+      <div
+        className={`flex flex-col p-5 after:absolute after:bottom-1 after:h-[1px] after:w-[calc(100%-40px)] after:bg-[rgba(0,0,0,0.1)] md:line-clamp-3 md:pt-[12px] md:after:hidden ${
```
bg-color 應為 design system 的顏色

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/article-card.tsx:
```diff
@@ -0,0 +1,126 @@
+const shouldShowComments = (category?: TabCategory) => {
+  if (category === TabCategory.BOOKMARKS) return false
+  return true
+}
```
`isCommentShow = category !== TabCategory.BOOKMARKS`

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/article-card.tsx:
```diff
@@ -0,0 +1,126 @@
+
+export type StoryItem = NonNullable<PickList>[number]['story']
+type ArticleCardProps = {
+  data: NonNullable<StoryItem>
```
建議 naming 可用 `story` or `storyData` ...etc

tyc67 commented on packages/mesh/app/profile/_components/comment/useCommentLogic.tsx:
```diff
@@ -0,0 +1,68 @@
+  route: string
+}
+
+export const useCommentLogic = (
```
這個custom hook 是不是放在 `hooks/`

tyc67 commented on packages/mesh/app/profile/_components/comment/useCommentLogic.tsx:
```diff
@@ -0,0 +1,68 @@
+  }, [clampLineCount])
+
+  const handleToggleClamp = useCallback(() => {
+    if (!commentRef.current) return
```
`if (!commentRef.current || !isTooLong) return`

LinHeMa committed:
feat(profile): header

LinHeMa committed:
refactor(profile): pr advices

LinHeMa committed:
refactor(profile): merge components

LinHeMa committed:
refactor(profile): delete unused condition

LinHeMa committed:
refactor(profile): publisher use customId

1. use publishers because publisher cannot filter by customId

tyc67 commented on packages/mesh/hooks/use-comment-logic.tsx:
```diff
@@ -0,0 +1,68 @@
+      router.push(shouldRedirect.route)
+      return
+    }
+    if (!isTooLong) return
```
Ln.33 有了, 這邊記得刪掉

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/page.tsx:
```diff
@@ -0,0 +1,62 @@
+}
+const page = async ({ params }: PageProps) => {
+  const publisherId = params.publisherId
+  console.log(publisherId)
```
remove debug `console.log()`

tyc67 commented on packages/mesh/app/profile/_components/article-card.tsx:
```diff
@@ -0,0 +1,124 @@
+
+export type StoryData = NonNullable<PickList>[number]['story']
+type ArticleCardProps = {
+  data: NonNullable<StoryData>
```
variable naming `story` ,`storyData` ...etc

tyc67 commented on packages/mesh/app/profile/_components/user-profile.tsx:
```diff
@@ -0,0 +1,41 @@
+        <Avatar size="xl" src={avatar} extra="sm:w-20 sm:h-20 object-cover" />
+        <div className="flex flex-col justify-center gap-1">
+          <p className="profile-title  sm:title-1 text-primary-700">{name}</p>
+          {userType !== 'publisher' && (
+            <p className="footnote sm:profile-subtitle text-primary-500">
+              本週精選了
+              <span className="text-primary-800"> {pickCount} </span>
+              篇文章
+            </p>
+          )}
```
是否少了 publisher 的情境 ? [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=6140-143687&t=uSiJ6hnjLe6GuxdE-4)

LinHeMa replied:
這個詳見 [slack_mesh](https://mirrormedia.slack.com/archives/C0716CSQNVD/p1719388146534159)

tyc67 replied:
PM 說保留樣式? [slack](https://mirrormedia.slack.com/archives/C0716CSQNVD/p1720755161185369?thread_ts=1719388146.534159&cid=C0716CSQNVD) , [PRD](https://paper.dropbox.com/doc/READr-Mesh-Web-PRD-WIP-pcFIPpEQo3Xf6ZoaQ0AJp#:uid=741011064233314067498508&h2=3.6-%E5%80%8B%E4%BA%BA%E6%AA%94%E6%A1%88)

LinHeMa replied:
先新增假資料（以回報在slack與PM確認）

tyc67 commented on packages/mesh/app/profile/publisher/[publisherId]/_component/publisher-page.tsx:
```diff
@@ -0,0 +1,52 @@
+  storyData,
+  userType,
+}) => {
+  const userStatusList = [{ tabName: TabKey.SPONSORED, count: '9999次' }]
```
是否少了 FOLLOWER ?

tyc67 commented on packages/mesh/app/profile/_components/user-status-list.tsx:
```diff
@@ -0,0 +1,21 @@
+        key={tabName}
+        className="relative flex h-[38px] w-[84px] flex-col items-center justify-between after:absolute after:-right-3 after:top-[9px] after:h-5 after:w-[0.5px] after:bg-primary-200 last:after:hidden sm:w-fit sm:flex-row sm:gap-1 sm:after:hidden"
+      >
+        <p className="profile-title-2 text-primary-700">{count}</p>
```
如果是 count = 990000 是否應該是顯示 99.9萬, [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=6140-139734&t=uSiJ6hnjLe6GuxdE-4)

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,94 @@
+      <Tab category={category} setCategory={setCategory} userType={userType} />
+      <ArticleCardList
+        picksOrBookmarks={picksOrBookmarks}
+        memeberId={memeberId}
```
memberId

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,94 @@
+  userType: userType
+  picksData: Picks
+  bookmarks: Bookmarks
+  memeberId: string
```
memberId

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/visitor-page.tsx:
```diff
@@ -0,0 +1,74 @@
+      <Tab category={category} setCategory={setCategory} userType={userType} />
+      <ArticleCardList
+        picksOrBookmarks={picksData}
+        memeberId={memberId}
```
memberId

tyc67 commented on packages/mesh/app/profile/member/[customId]/page.tsx:
```diff
@@ -0,0 +1,97 @@
+          userType={userType}
+          picksData={picksData}
+          bookmarks={bookmarkData}
+          memeberId={userData.id}
```
memberId

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/article-card-list.tsx:
```diff
@@ -0,0 +1,70 @@
+}
+const ArticleCardList = ({
+  picksOrBookmarks,
+  memeberId,
```
memberId

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/visitor-page.tsx:
```diff
@@ -0,0 +1,74 @@
+      <div className="flex max-h-[calc(100%_-_152px)] flex-col items-center bg-white px-5 pb-8 pt-6 sm:max-h-full">
+        <UserProfile
+          name={name}
+          pickCount={pickCount || 0}
```
`<Page/>` 已經做null check

tyc67 commented on packages/mesh/app/profile/member/[customId]/_components/member-page.tsx:
```diff
@@ -0,0 +1,94 @@
+      <div className="flex max-h-[calc(100%_-_152px)] flex-col items-center bg-white px-5 pb-8 pt-6 sm:max-h-full">
+        <UserProfile
+          name={name}
+          pickCount={pickCount || 0}
```
`<Page/>` 已經做null check

tyc67 commented on packages/mesh/app/profile/_components/tab.tsx:
```diff
@@ -0,0 +1,79 @@
+
+type TabItem = {
+  name: string
+  category: TabCategory
```
naming 建議用 `TabCategory` or `TabOOXX` ...etc, 避免與 CMS 的 Category 混淆

LinHeMa committed:
refactor(profile): naming

LinHeMa committed:
refactor(profile): add picked count mock data

LinHeMa committed:
refactor(profile): layout

LinHeMa committed:
refactor(profile): merge 2 components

LinHeMa committed:
refactor(profile): naming

tyc67 commented on packages/mesh/app/profile/_components/article-card-list.tsx:
```diff
@@ -0,0 +1,44 @@
+
+interface ArticleCardListProps<CardItemType> {
+  items: CardItemType[]
+  renderItem: (
+    item: CardItemType,
+    index: number,
+    isLast: boolean
+  ) => React.ReactNode
```
為何採用 React.Node,而不是寫在Component 裡裏? React.Node 的開放性在這裏是否有其他的用途?

LinHeMa replied:
我可能會需要了解你的問題，我這邊處理的原因是因為story的資料節點跟pick不同，所以讓render function可以custom

tyc67 replied:
在 `<ArticleCardList/>` 有較開放性的 items and renderItem 設計, 然目前只有 `<ArticleCard />` 使用情境,  在 props property 邏輯上我個人會選擇, 給你參考

```javascript
interface ArticleCardListProps {
  items: StoryData | Picks | Bookmarks
  userType: string
  emptyMessage: string
  category?: TabCategory
  memberId?: string
  avatar?: string
  name?: string
}

      {items.map((item, index) => {
        const isLast = index === items.length - 1
        return (
          <li
            key={index}
            className="relative w-full bg-white md:h-full md:rounded-md md:drop-shadow"
          >
            items === storydata ? (
              <ArticleCard
                storyData={item}
                isLast={isLast}
                userType={userType}
              />
            ) : (
              <ArticleCard
                storyData={item.story}
                isLast={isLast}
                memberId={memberId}
                avatar={avatar}
                category={category}
                name={name}
                userType={userType}
              />
            )}
          </li>
        )
      })}

```

tyc67 commented on packages/mesh/app/profile/_components/tab.tsx:
```diff
@@ -0,0 +1,81 @@
+      {tabList
+        .filter(tabFilter)
+        .map(({ tabName, tabCategory, tabActiveStyle }) => (
+          <li
+            key={tabName}
+            onClick={() => handleTabOnClick(tabCategory)}
+            className={`button-large relative cursor-pointer p-[14px] pt-3 ${tabActiveStyle} sm:px-8 sm:py-[13px]`}
+          >
```
`${tabCategory === tabCategory ? activeTabStyle : inactiveTabStyle}`

LinHeMa replied:
想問換成下面的寫法是有什麼技術上的建議嗎？
還是說是比較屬於偏好的部分呢？

caesarWHLee replied:
插個話，原本的寫法 `tabActiveStyle` 應該改成 `tabStyle`，因為每個 tab 都有可能是 `activeStyle` 或是 `inactiveStyle`

tyc67 replied:
> 想問換成下面的寫法是有什麼技術上的建議嗎？ 還是說是比較屬於偏好的部分呢？

已經有 props 已有 `tabCategory` state, 可以不用另外使用 `tabActiveStyle` 去做 condition , 如果 tablist 中的 tabCategory === 當前的 tabCategory state, 應該就是 activeTabStyle, 若不等於則是 inactiveTabStyle

tyc67 commented on packages/mesh/app/profile/_components/user-profile.tsx:
```diff
@@ -0,0 +1,55 @@
+import Avatar from '@/components/story-card/avatar'
+
+type userType = 'member' | 'visitor' | 'publisher'
```
各 component 的 type UserType, 來源應統一, 建議可以整理一下

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,189 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+        }
+        title
+        og_title
+        commentCount
```
`commentCount` 重複出現

caesarWHLee commented on packages/mesh/graphql/query/member.gql:
```diff
@@ -100,3 +90,189 @@ query GetMemberByFollowingCategory($slugs: [String!]!) {
+        }
+        title
+        og_title
+        commentCount
```
`commentCount` 重複出現

caesarWHLee commented on packages/mesh/app/profile/_components/article-card.tsx:
```diff
@@ -0,0 +1,124 @@
+            <StoryPickButton isStoryPicked={false} storyId={storyData.id} />
+          </div>
+        </section>
+        {isCommentShow && (
```
建議：與其把 `category` 和 `userType` 傳進 `ArticleCard`，不如改成傳 `showComment` (`shouldShowComment`, `isShowComment` 都可)，由三個 page 層去決定要不要 show comment，這樣 `PublisherPage` 就直接帶 `showComment={false}` 而剩下兩個 page 再依照 TabCategory 來決定是否 show comment，日後維護其中一個 page 的時候不需要知道所有 pages 的 comment 顯示邏輯，可以專注在要改的功能上面

caesarWHLee commented on packages/mesh/app/profile/_components/article-card.tsx:
```diff
@@ -0,0 +1,124 @@
+  userType,
+}: ArticleCardProps) => {
+  const commentList = storyData.comment || []
+  const authorComment =
```
這邊好像沒有處理到 comment 要是該 member 的才算，如果是之後才會補上的功能建議加上 TODO comment

caesarWHLee commented on packages/mesh/hooks/use-comment-logic.tsx:
這個 hook 應該只要處理 clamp style 邏輯就好，redirect 邏輯應該直接拆出來在 comment 的 `onClick` function 中直接處理就好

`use-comment-logic` -> `use-comment-clamp`

caesarWHLee commented on packages/mesh/app/profile/_components/comment.tsx:
```diff
@@ -0,0 +1,87 @@
+        className={`relative md:flex md:items-start ${
+          needClamp ? '' : 'after:opacity-0'
+        } after:body-3 after:absolute after:bottom-0 after:right-1 after:bg-gradient-to-r after:from-transparent after:from-0% after:to-primary-100 after:to-25% after:pl-6 after:text-primary-400 after:content-['...繼續閱讀'] md:after:bottom-[6px]`}
+        onClick={handleToggleClamp}
```
搭配 `use-comment-logic` 的 comment，這邊應該直接在用一個 callback function 來處理 onClick，把 redirect 和 use-comment-clamp 邏輯區隔

```
onClick={() => {
  if (width > 960) {
     // hadle redirect
  }else {
     // handle toggle clamp 
  }
}}
```

LinHeMa committed:
refactor(profile): delete duplicate

LinHeMa committed:
refactor(profile): pr message

tyc67 commented on packages/mesh/types/profile.ts:
```diff
@@ -0,0 +1,30 @@
+export type Bookmarks = NonNullable<GetMemberProfileQuery['member']>['books']
+export type StoryData = NonNullable<GetPublisherProfileQuery['stories']>
+export type PickListItem = NonNullable<PickList>[number]['story']
+export type userType = 'member' | 'visitor' | 'publisher'
```
應為 "U"serType

tyc67 commented on packages/mesh/app/profile/_components/user-profile.tsx:
```diff
@@ -0,0 +1,54 @@
+  return (
+    <>
+      <section className="flex w-full gap-4">
+        <Avatar size="xl" src={avatar} extra="sm:w-20 sm:h-20 object-cover" />
```
媒體個人檔案的avatar形式似乎不同 ? , [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=6140-139734&t=lCWT2xGGTcBu7UdC-4)

tyc67 commented on packages/mesh/app/profile/_components/article-card-list.tsx:
```diff
@@ -1,22 +1,25 @@
-  ) => React.ReactNode
-  userType: string
-  category?: TabCategory
+type StoryData = any // 請根據實際類型定義替換
```
import from '@/types/profile' ?

tyc67 commented on packages/mesh/app/profile/_components/article-card.tsx:
```diff
@@ -69,28 +60,28 @@ const ArticleCard = ({
-                paywall={storyData.paywall || false}
-                fullScreenAd={storyData.full_screen_ad || ''}
+                commentCount={storyData?.commentCount || 0}
+                publishDate={storyData?.published_date || Date.now()}
```
publish 時間應該不會是 `Date.now()` ?

caesarWHLee commented on packages/mesh/app/profile/_components/article-card.tsx:
```diff
@@ -0,0 +1,115 @@
+  name,
+  shouldShowComment,
+}: ArticleCardProps) => {
+  // 此處在gql部分只會拿最新、且是個人檔案本人的留言。
```
註解建議可以把脈絡寫完整一點 (補充 publisher 沒有 comment 的情境)，publisher 的 story 不會有 comment， member 的 pick/bookmark story comment 會特別拿該個人檔案帳號的 comment

caesarWHLee commented on packages/mesh/app/profile/_components/comment.tsx:
```diff
@@ -0,0 +1,92 @@
+  clampLineCount = 3,
+  avatar,
+  canToggle = false,
+  // 之後有文章在更改成slug或id傳入做跳轉功能。
```
註解可以補一下 `TODO:` prefix 以便後續查找

caesarWHLee commented on packages/mesh/app/profile/_components/comment.tsx:
```diff
@@ -0,0 +1,92 @@
+    canToggle
+  )
+  const handleCommentClick = () => {
+    if (width > 960) {
```
這邊和下面的 960 寬度都建議改成從 tailwindConfig 中的 breakpoint 去拿來統一 breakpoint 使用

可以擴充 `@/utils/tailwind.ts`  如下 (或是改成你習慣的其他寫法)：

```
import resolveConfig from 'tailwindcss/resolveConfig'

import tailwindConfig from '@/tailwind.config'

type Breakpoint = 'sm' | 'md' | 'lg' | 'xl' | '2xl'

function getTailwindConfig() {
  return resolveConfig(tailwindConfig)
}

function getTailwindConfigBreakpointNumber(breakpoint: Breakpoint) {
  const tailwindConfig = getTailwindConfig()
  return parseInt(tailwindConfig.theme.screens[breakpoint]
}

export { getTailwindConfig, getTailwindConfigBreakpointNumber }
```

並把 960 改成 getTailwindConfigBreakpointNumber('md') 的方式

caesarWHLee commented on packages/mesh/graphql/query/publisher.gql:
```diff
@@ -8,3 +8,47 @@ query Publishers {
+      title
+      official_site
+    }
+    title
```
重複的 title

caesarWHLee reviewed (commented):
是說 comment 其實要留給 reviewer 去按 resolved，方便讓 reviewer 確認當初留了什麼 comment XD

LinHeMa committed:
refactor(probile): change userType

LinHeMa committed:
refactor(profile): avatar, comment

caesarWHLee reviewed (approved):
LGTM 👍
